pkg/controller/pod/pod_controller.go

```diff
@@ -250,3 +264,106 @@ func (r *ReconcilePod) createK8sObject(obj *unstructured.Unstructured) error {
 	}
 	return nil
 }
+
+func (r *ReconcilePod) recreateNodeAgentPodsIfInvalidResolvConf(
+	resName string) error {
+	podsInCLB, err := identifyPodsInCLBDueToInvalidResolvConf(r.client)
+	if err != nil {
+		log.Error(err, "Could not identify if any pod is in CLB because "+
+			"of invalid resolv.conf")
+		return err
+	}
+	if len(podsInCLB) > 0 && !deletePods(podsInCLB, r.client) {
+		err := errors.New("Error occured while trying to restart pods in " +
+			"CLB because of invalid resolv.conf")
+		log.Error(err, "")
+		return err
+	}
+	return nil
+}
+
+func identifyPodsInCLBDueToInvalidResolvConf(c client.Client) (
+	[]corev1.Pod, error) {
+	var podsInCLB []corev1.Pod
+	podList := &corev1.PodList{}
+	nodeAgentLabelSelector := labels.SelectorFromSet(
+		map[string]string{"component": operatortypes.NsxNodeAgentDsName})
+	err := c.List(context.TODO(), podList, &client.ListOptions{
+		LabelSelector: nodeAgentLabelSelector})
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range podList.Items {
+		if isNodeAgentContainerInCLB(&pod) {
+			nodeAgentLogs, err := getContainerLogsInPod(
+				&pod, operatortypes.NsxNodeAgentContainerName)
```

Comment:
This can retrieve quite a big of data, and if the pod is in CLBO, will we call quite often?

Reply:
good point. let me check if we just get the last 100 lines or so

Reply:
fetching last 50 log lines

Reply:
awesome!